crates/editor/src/editor.rs

```diff
@@ -4110,6 +4110,39 @@ impl Editor {
                 active_index: 0,
                 ranges: tabstops,
             });
+
+            // Gets whether the currently entered snipper ends with a surround
+            let snapshot = self.buffer.read(cx).snapshot(cx);
+            let selection = self.selections.first::<Point>(cx);
+            let start_column = snapshot.indent_size_for_line(selection.start.row).len;
+            let scope = snapshot
+                .language_scope_at(Point::new(selection.start.row, start_column))
+                .unwrap();
+            let mut bracket_pair = None;
+            for (pair, enabled) in scope.brackets() {
+                if enabled && pair.close && snippet.text.ends_with(pair.end.as_str()) {
+                    bracket_pair = Some(pair.clone());
+                    break;
+                }
+            }
+
+            // If it ends with an available surround, insert the auto close region
+            if self.autoclose_regions.is_empty() {
+                if let Some(pair) = bracket_pair {
+                    let start = snapshot
+                        .anchor_after(Point::new(selection.start.row, selection.start.column));
+                    let end =
+                        snapshot.anchor_after(Point::new(selection.end.row, selection.end.column));
+                    self.autoclose_regions.insert(
+                        0,
+                        AutocloseRegion {
```

Comment:
This can be a `.push`, since `self.autoclose_regions` is a `Vec` and we checked that it's empty anyway:

```suggestion
                    self.autoclose_regions.push(AutocloseRegion {
```